File: tools/journeys/src/journeys/report.py
"""报告序列化：JSON 到 stdout 或 --out 文件（spec §3.5）。"""

import json
from pathlib import Path

def emit(report, out_path=None):
    text = json.dumps(report, ensure_ascii=False, indent=2)
    if out_path:
        Path(out_path).write_text(text + "\n", encoding="utf-8")
    else:
        print(text)

File: tools/journeys/internal/journeys/report.go
package journeys

import (
	"encoding/json"
	"io"
	"os"
)

// Emit 将报告序列化为两空格缩进 JSON：outPath 非空写入该文件，否则打印到 stdout。
func Emit(rep *Report, outPath string, stdout io.Writer) error {
	data, err := json.MarshalIndent(rep, "", "  ")
	if err != nil {
		return err
	}
	data = append(data, '\n')
	if outPath != "" {
		return os.WriteFile(outPath, data, 0o644)
	}
	_, err = stdout.Write(data)
	return err
}
